mpdsync.py

```diff
@@ -48,9 +48,15 @@ class MyFloat(float):
     # float, you have to also override those built-in methods to keep
     # returning the subclass.
 
-    def __init__(self, num, roundBy=3):
-        super(MyFloat, self).__init__(num)
-        self.roundBy = roundBy
+    if sys.version_info[0] < 3:
+        def __init__(self, num, roundBy=3):
+            super(MyFloat, self).__init__(num)
+            self.roundBy = roundBy
+    else:
+        def __new__(cls, num, roundBy=3):
+            instance = super().__new__(cls, num)
+            instance.roundBy = roundBy
+            return instance
```

Comment:
Thanks for updating the call to `super()`, but why is it necessary to use `__new__` instead of `__init__`?

Reply:
I reverted the change and tested again under Python 3, and confirmed initialising the MyFloat using __init__ causes a TypeError.

> ERROR: mpdsync: Unable to connect to master server: object.__init__() takes exactly one argument (the instance to initialize)
Traceback (most recent call last):
  File "/home/sambr/mpdsync/mpdsync.py", line 1343, in main
    master.connect()
  File "/home/sambr/mpdsync/mpdsync.py", line 277, in connect
    self.testPing()
  File "/home/sambr/mpdsync/mpdsync.py", line 477, in testPing
    self.ping()
  File "/home/sambr/mpdsync/mpdsync.py", line 225, in ping
    self.pings.insert(0, timeFunction(super(Client, self).ping))
  File "/home/sambr/mpdsync/mpdsync.py", line 124, in insert
    args = [MyFloat(a) for a in args]
            ^^^^^^^^^^
  File "/home/sambr/mpdsync/mpdsync.py", line 52, in __init__
    super(MyFloat, self).__init__(num)
TypeError: object.__init__() takes exactly one argument (the instance to initialize)